src/app/wallets/update-on-chain-receipt.ts

```diff
@@ -169,15 +178,27 @@ const processTxForWallet = async (
           })
           if (walletAddressReceiver instanceof Error) return walletAddressReceiver
 
-          const feeDisplayCurrency = Number(
-            walletAddressReceiver.usdBankFee.amount,
-          ) as DisplayCurrencyBaseAmount
+          const recipientAccount = await AccountsRepository().findById(wallet.accountId)
+          if (recipientAccount instanceof Error) return recipientAccount
+          const { displayCurrency } = recipientAccount
 
+          const recipientDisplayPriceRatio = await getCurrentPriceAsDisplayPriceRatio({
+            currency: displayCurrency,
+          })
+          if (recipientDisplayPriceRatio instanceof Error) {
+            return recipientDisplayPriceRatio
+          }
           const amountDisplayCurrency = Number(
-            walletAddressReceiver.usdToCreditReceiver.amount,
+            recipientDisplayPriceRatio.convertFromWallet(
+              walletAddressReceiver.btcToCreditReceiver,
+            ).amountInMinor,
           ) as DisplayCurrencyBaseAmount
```

Comment:
This unpacking/casting will be cleaned up in a follow-up PR together with the remnants from #2384